rebasehelper/build_log_hooks/files_build_log_hook.py

```diff
@@ -0,0 +1,241 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import difflib
+import fnmatch
+import os
+import re
+
+import six
+
+from rebasehelper.build_log_hook import BaseBuildLogHook
+from rebasehelper.helpers.macro_helper import MacroHelper
+from rebasehelper.logger import logger
+
+
+class FilesBuildLogHook(BaseBuildLogHook):
+    NAME = 'files'
+    CATEGORIES = None
+
+    # taken from build/files.c in RPM source
+    FILES_DIRECTIVES = [
+        '%artifact',
+        '%attr',
+        '%caps',
+        '%config',
+        '%defattr',
+        '%dev',
+        '%dir',
+        '%doc',
+        '%docdir',
+        '%dverify',
+        '%exclude',
+        '%ghost',
+        '%lang',
+        '%license',
+        '%missingok',
+        '%pubkey',
+        '%readme',
+        '%verify',
+    ]
+
+    @classmethod
+    def get_name(cls):
+        return cls.NAME
+
+    @classmethod
+    def get_categories(cls):
+        return cls.CATEGORIES
+
+    @classmethod
+    def _parse_build_log(cls, log_path, nvr):
+        """Parses a build log.
+
+        Args:
+            log_path (str): Path to the RPM build log.
+
+        Returns:
+            tuple: The first element is the type of error that was found
+                in the log (missing or deleted). The second element is
+                a list of problematic files.
+
+        """
+        try:
+            with open(log_path, 'r') as build_log:
+                lines = build_log.read().splitlines()
+        except IOError:
+            logger.error('There was an error opening %s', log_path)
+            return None, None
+
+        error_type = None
+        # Use set to avoid duplicate files.
+        files = set()
+
+        # Two types of error can occur. The file is either in the SPEC file but missing in sources
+        # or the other way around. Example lines of the first case:
+        # File not found: /root/rpmbuild/BUILDROOT/pello-0.1.1-1.fc27.x86_64/should/not/be/here
+        # File not found: /root/rpmbuild/BUILDROOT/pello-0.1.1-1.fc27.x86_64/should/not/be/here2
+        # Example lines of the second case:
+        # Installed (but unpackaged) file(s) found:
+        #   /usr/bin/pello
+        error_re = re.compile(
+            r'''
+            ^
+            (BUILDSTDERR:)?
+            \s*
+            (
+            (?P<missing>File not found:\s*)|
+            (?P<unpackaged>Installed \(but unpackaged\) file\(s\) found:)
```

Comment:
Spaces are ignored in verbose regexps, you should use `[ ]`, or `\ ` or more general `\s`.